client/src/components/FrontPage.js

```diff
@@ -10,33 +10,38 @@ import BasicGrid from './front-page/BasicGrid';
 import CourseTable from './front-page/CourseTable';
 import coursesService from '../services/courses';
 import { useNavigate } from 'react-router-dom';
+import useAuth from '../hooks/useAuth';
 
 const FrontPage = () => {
   const navigate = useNavigate();
   const [currentCourses, setCurrentCourses] = useState([]);
   const [previousCourses, setPreviousCourses] = useState([]);
 
-  useEffect(() => {
-    coursesService.getCourses()
-      .then((data) => setCurrentCourses(data.courses.current))
-      .catch((e) => console.log(e.message));
-  }, []);
+  const { auth } = useAuth();
 
   useEffect(() => {
     coursesService.getCourses()
-      .then((data) => setPreviousCourses(data.courses.previous))
+      .then((data) => {
+        setCurrentCourses(data.courses.current);
+        setPreviousCourses(data.courses.previous);
+      })
       .catch((e) => console.log(e.message));
   }, []);
 
+  // Conditional rendering: only admin and teachers are shown the button for creating a new course
```

Comment:
The commenting on this PR is very good! So, could this comment be of the same format as the others between components? As an example: CourseView.js, line 35